sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/BodyMemory.cs

```diff
@@ -6,96 +6,144 @@
 using System.Collections.Generic;
 using System.Linq;
 using Azure.Core;
-using Microsoft.Azure.Amqp.Encoding;
 using Microsoft.Azure.Amqp.Framing;
 
 namespace Azure.Messaging.ServiceBus.Amqp
 {
-    internal sealed class BodyMemory : IEnumerable<ReadOnlyMemory<byte>>
+    internal abstract class BodyMemory : IEnumerable<ReadOnlyMemory<byte>>
```

Comment:
I'd really appreciate some comments/doc comments.   We're at a level that is reasonably complex, fairly dense, and is not likely to be something that gets revisited often.   

Understanding even just the high level intent and flow would be very helpful for those who will be maintaining or potentially looking to apply something similar elsewhere.

Reply:
Added some comments and documentation. By the way if you want to do small tweaks I gave edit rights on this PR.